File: backend/Models/AssignRole.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace backend.Models
{
    [Table("AssignRole")]
    public class AssignRole : BaseModel
    {
        [Key]
        public int Id { get; set; }
        public string? EmployeeId { get; set; }
        public string Employee { get; set; }
        public string? Role { get; set; }
        public int? RoleId { get; set; }
        public string? Department { get; set; }
        public DateTime? StartDate { get; set; }
        public DateTime? EndDate { get; set; }
        public string? Remarks { get; set; }
    }
}

File: frontend/src/app/master/models/assignRole.ts
export interface AssignRole {
  id?: number;
  role: string;
  employeeId: string;
  employee: string;
  department: string;
  remarks?: string;
  startDate: string | null;
  endDate: string | null;
  active: boolean;
  status: number;
  createdOn: string;
  createdBy: string;
  modifiedOn: string;
  modifiedBy: string;
  roleId?: number;
  employees: EmployeeDto[];
  employeeCount: number;
  deptId: string;
}

export interface EmployeeDto {
  employeeID: string;
  employeeName: string;
}